base/intfuncs.jl

```diff
@@ -82,14 +82,16 @@ end
 
 gcd(a::Union{Integer,Rational}) = a
 lcm(a::Union{Integer,Rational}) = a
-gcd(a::Union{Integer,Rational}, b::Union{Integer,Rational}) = gcd(promote(a,b)...)
-lcm(a::Union{Integer,Rational}, b::Union{Integer,Rational}) = lcm(promote(a,b)...)
-gcd(a::Union{Integer,Rational}, b::Union{Integer,Rational}...) = gcd(a, gcd(b...))
-lcm(a::Union{Integer,Rational}, b::Union{Integer,Rational}...) = lcm(a, lcm(b...))
+gcd(a::Real, b::Real) = gcd(promote(a,b)...)
+lcm(a::Real, b::Real) = lcm(promote(a,b)...)
+gcd(a::Real, b::Real...) = gcd(a, gcd(b...))
```

Comment:
This should probably be `gcd(a::Real, b::Real, c::Real...)` to avoid trying to call `gcd()` --- it would be an error either way but clearer to get a method error for `gcd(x::T)` where the implementation for `T` is missing.

Reply:
I agree, will change it.